OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -0,0 +1,243 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public class ActorEditLogic : ChromeLogic
+	{
+		readonly World world;
+		readonly WorldRenderer worldRenderer;
+		readonly EditorActorLayer editorActorLayer;
+		readonly EditorViewportControllerWidget editor;
+		readonly ContainerWidget actorSelectBorder;
+		readonly BackgroundWidget actorEditPanel;
+		readonly LabelWidget typeLabel;
+		readonly TextFieldWidget actorIdField;
+		readonly DropDownButtonWidget ownersDropDown;
+		readonly LabelWidget validationLabel;
+		readonly int originalHeight; // Height of the edit widget without the validation label showing.
+		readonly int validationHeight; // Height of the edit widget with the validation label showing.
+		readonly int editPanelPadding; // Padding between right edge of actor and the edit panel.
+		readonly TimeSpan scrollVisibleTimeout
```

Comment:
We use `Game.RunTime` or `World.WorldTick` everywhere else in the game code for evaluating time.  It would be better if we could use one of these here instead of introducing a third method.